### ISSUE TITLE
Have bazel generate API changes

### ISSUE THREAD

**Issue by chrislovecnm** [kubernetes/kops#3695]
Should we have this? https://github.com/kubernetes/kubernetes/blob/master/pkg/generated/openapi/def.bzl#L27

**Comment by johngmyers** [comment]
Referenced link is dead. Would need more information.

/remove-lifecycle frozen

**Comment by chrislovecnm** [comment]
@johngmyers what tool are we currently using to generate the code when we make API changes?

**Comment by johngmyers** [comment]
k8s.io/code-generator v0.18.1

**Comment by chrislovecnm** [comment]
This jetstack repo is running code-generator bazel here https://github.com/jetstack/cert-manager/blob/master/hack/BUILD.bazel#L148 - They also have a bazel test verifying that the API has been generated.  The command is a `bazel run` not part of the build phase which is not how I would like it.  If you can tell me how to get it as part of the build phase, that would be AWESOME.

I have not looked at kops build in awhile to see if bazel is generating and verifying the APIs when building kops.

**Comment by johngmyers** [comment]
A developer changing the API invokes `make apimachinery` and commits the result. There are Prow and Github Actions jobs which run a make target to verify the generation result matches what was committed.

I don't see the particular need or advantage of doing this through Bazel.

**Comment by chrislovecnm** [comment]
I have had problems, in the past, with the wrong version of API machinery being picked up, instead of the tagged version.  This was a while back, so it may have been fixed since then.  Using the code I showed you ensure that the build is deterministic. Also, I have run into problems where the go version that I have installed locally is not the go version that I should use.  Using bazel for this addresses both of those problems that I have faced in the past.

As I mentioned those problems may have been fixed since I ran into them, and this ticket was opened a long time ago.  Is there a bazel target where I run code generation?
